p2p/address-book/src/peer_list.rs

```diff
@@ -104,25 +104,21 @@ impl<Z: NetworkZone> PeerList<Z> {
                 })?;
                 let n = r.gen_range(0..addresses_with_block.len());
                 let peer = addresses_with_block[n];
-                if must_keep_peers.contains(&peer) {
-                    continue;
+                if !must_keep_peers.contains(&peer) {
+                    return self.remove_peer(&peer);
                 }
```

Comment:
Ah, I didn't edit https://github.com/Cuprate/cuprate/pull/118#discussion_r1585534711 in time I guess - I introduced a bug here, it should be reverted.

If this doesn't `continue` then the code below will execute, this is different behavior than before. Should be:
```rust
if must_keep_peers.contains(&peer) {
    continue;
}

return self.remove_peer(&peer);
```

Reply:
Ah, good spot!